### PR TITLE
Test/user updates

### PR DESCRIPTION
**Pull request by mindthespines**
There are the tests I originally set out to do, plus a whole mess of general housekeeping in the users/@me testing area that may make this PR hard to parse out - please let me know if you'd like me to make separate PRs! (one with the tests and one with the organization)

Summary:
- Adds tests for the `/users/@me` PATCH endpoint
- Adds tests for the `/users/@me/settings` GET and PATCH endpoints
- Organizes tests for the `/users/@me/...` endpoints by endpoint; moves a stray query test to its own file

...I think that covers it? There are a couple questions/notes I'll put in the comments. The big question though, since the tests are split up differently, is whether the correct things are being mocked in each file. I'd say keep an eye out for excess or missing mocks in the new files! The tests all pass as they are so I think it's set up fine but am not entirely confident.

### CODE REVIEW

File: server/users/routes.ts [hunk 1]
```diff
@@ -376,6 +376,7 @@ router.patch("/@me/settings", ensureLoggedIn, async (req, res) => {
 
     res.status(200).json(aggregateByType(settings));
   } catch (e) {
+    // TODO: there are three different awaits in the try block, only one whose failure would suggest the settings weren't updated, consider adjusting this message
```
Comment (mindthespines) [inline server/users/routes.ts:379]: This is a question I had; didn't want to update this message without checking - can this be changed to something more general like "Something went wrong while updating settings?" At a glance it looks possible that settings might update but then something goes wrong in the subsequent tasks.

File: server/users/tests/routes/user-data.test.ts [hunk 2]
```diff
@@ -1,57 +1,121 @@
...
-    expect(res.body.pinned_boards).toEqual({});
+  test("prevents unauthorized PATCH access to the @me endpoint", async () => {
+    await wrapWithTransaction(async () => {
+      const res = await request(server.app).patch(`/@me`).send(testUserPatch);
+
+      expect(res.status).toBe(401);
+    });
   });
 
-  // TODO: check if any test user has outdated notifications or none, and test there
```
Comment (mindthespines) [inline server/users/tests/routes/user-data.test.ts:54]: this same TODO exists in the notifications tests, figured it was okay to remove

File: server/users/tests/routes/user-data.test.ts [hunk 3]
```diff
@@ -1,57 +1,121 @@
...
+  });
+
+  test("returns a bad request error if username or avatar URL is missing", async () => {
+    await wrapWithTransaction(async () => {
+      setLoggedInUser(JERSEY_DEVIL_USER_ID);
+      const res = await request(server.app)
+        .patch(`/@me`)
+        .send({ username: "Newname" });
 
-// TODO: test dismiss notifications
```
Comment (mindthespines) [inline server/users/tests/routes/user-data.test.ts:57]: this test exists in notifications.test.ts, figured this was okay to remove

File: server/users/routes.ts (hunk 1 above)
Comment (essential-randomness) [reply server/users/routes.ts:379]: You're very much right, and it might make sense to just wrap the first `await` in the try catch with the custom message. I'm 95% sure if we get an error in the subsequent one, the server will automatically generate a 500 error anyway.

File: server/users/tests/routes/user-data.test.ts (hunk 2 above)
Comment (essential-randomness) [reply server/users/tests/routes/user-data.test.ts:54]: thank you!

**Review by essential-randomness — APPROVED** [review]
If you want to make the change to the user setting route, please do! If not, i'm ok merging this as is.

Thanks for writing all these awesome tests.

**Comment by mindthespines** [comment]
cool! I'll make an update in that settings route, pull in latest, and push back up.

**Comment by mindthespines** [comment]
this should be set now
